HW2/P5/driver.py

```diff
@@ -80,3 +100,27 @@ def randcolor():
             # SUBPROBLEM 3: sort objects by location.  Be sure to update the
             # grid if objects' indices change!  Also be sure to sort the
             # velocities with their object positions!
+
+            #sort positions by a space sorting function
+            mapped_positions = zip([i for i in range(len(positions))], positions)            
+            mapped_positions = map(lambda (i,(x,y)): (i, (int(x/grid_spacing), int(y/grid_spacing))), mapped_positions)            
+            mapped_positions = sorted(mapped_positions, cmp=cmp_zorder)
+
+            #get the list of indices
+            position_indices = map(lambda (i,(x,y)): i, mapped_positions)
+
+            #re-sort positions and velocities for these indices
+            positions = positions[position_indices]
+            velocities = velocities[position_indices]
+
+            #reset the grid
+            grid = - np.ones((grid_size, grid_size), dtype=np.uint32)
+
+            #update all grid indices     
+            for i in range(num_balls):
+                grid_x_bucket = int(positions[i][0]/grid_spacing)
+                grid_y_bucket = int(positions[i][1]/grid_spacing)
+
+                #check if the bounds are in range
```

Comment:
I really like what you did here, it is much simpler than what I did, possible also much more efficient. Great job!